File: android-kit-base/src/androidTest/java/com/mparticle/kits/BaseTestKit.java
package com.mparticle.kits;

import android.content.Context;

import java.util.List;
import java.util.Map;

public class BaseTestKit extends KitIntegration {
    static OnKitCreateListener onKitCreateListener;

    @Override
    public String getName() {
        return null;
    }

    @Override
    protected List<ReportingMessage> onKitCreate(Map<String, String> settings, Context context) throws IllegalArgumentException {
        if (onKitCreateListener != null) {
            onKitCreateListener.onKitCreate(this);
            onKitCreateListener = null;
        }
        return null;
    }

    @Override
    public List<ReportingMessage> setOptOut(boolean optedOut) {
        return null;
    }

    public interface OnKitCreateListener {
        void onKitCreate(BaseTestKit kitIntegration);
    }

    public static void setOnKitCreate(OnKitCreateListener listener) {
        onKitCreateListener = listener;
    }
}

File: android-core/src/androidTest/kotlin/com/mparticle/kits/testkits/BaseTestKit.kt
package com.mparticle.kits.testkits

import android.content.Context
import com.mparticle.kits.KitIntegration
import com.mparticle.kits.ReportingMessage

open class BaseTestKit : KitIntegration() {
    open override fun onKitCreate(
        settings: Map<String, String>?,
        context: Context
    ): List<ReportingMessage> {
        return listOf()
    }

    open override fun setOptOut(optedOut: Boolean): List<ReportingMessage> {
        // do nothing
        return listOf()
    }

    open override fun getName(): String {
        return this::class.java.simpleName
    }

    open override fun getInstance() = this
}
